SystemInfo/Registry/IdentifierBased.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * File containing the IdentifierBased class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzSupportToolsBundle\SystemInfo\Registry;
+
+use EzSystems\EzSupportToolsBundle\SystemInfo\SystemInfoCollectorRegistry;
+
+/**
+ * A registry of SystemInfoCollectors that uses an identifier string to identify the collector.
+ */
+class IdentifierBased implements SystemInfoCollectorRegistry
+{
+    /** @var \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector[] */
+    private $registry = [];
+
+    /**
+     * Adds SystemInfoCollectors, each with an identifier string, to the registry.
+     *
+     * @param \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector[] $items A hash of SystemInfoCollectors, with identifier string as key.
+     */
+    public function addItems(array $items)
+    {
+        $this->registry = array_merge($this->registry, $items);
+    }
+
+    /**
+     * Returns the SystemInfoCollector matching the argument.
+     *
+     * @param string $identifier An identifier string.
+     *
+     * @return \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector|null The SystemInfoCollector, or null if there is none.
+     */
+    public function getItem($identifier)
+    {
+        if (isset($this->registry[$identifier])) {
+            return $this->registry[$identifier];
+        }
+
+        return null;
```

Comment:
I would make it throw an exception if the item doesn't exist. Do you think it is too much ? Would avoid the return value handling in the higher layers.